File: EggMoneyna/src/main/java/shinhan/EggMoneyna/monster/dto/MonsterResponseDto.java
package shinhan.EggMoneyna.monster.dto;


import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import shinhan.EggMoneyna.monster.entity.Monster;

@Getter
@NoArgsConstructor
public class MonsterResponseDto {
    private String name;

    private String status;

    private String feel;

    private int exp;

    private String benefit;

    @Builder
    public MonsterResponseDto(String name, String status, int point, String feel, int exp, String benefit) {
        this.name = name;
        this.status = status;
        this.feel = feel;
        this.exp = exp;
        this.benefit = benefit;
    }

}

File: android/app/src/main/java/com/shbhack/eggmoneyna/data/model/MonsterResponseDto.kt
package com.shbhack.eggmoneyna.data.model

data class MonsterResponseDto(
    val exp: Int,
    val status: String,
    val feel: String,
    val historyList: List<HistoryItem>
)

data class HistoryItem(
    val createTime: String,
    val updateTime: String,
    val id: Int,
    val content: String,
    val exp: Int,
    val prefix: Int,
    val monster: MonsterDetail
)

data class MonsterDetail(
    val createTime: String,
    val updateTime: String,
    val id: Int,
    val name: String,
    val status: String,
    val feel: String,
    val exp: Int,
    val benefit: String
)

